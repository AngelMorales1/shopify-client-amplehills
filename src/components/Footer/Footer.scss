@import 'settings', 'src/styles/typography';

.Footer {
  align-items: center;

  > span {
    @media ($medium-up) {
      margin-left: spacing(16);
    }
  }

  @media ($medium-up) {
    align-items: flex-start;
    position: relative;
  }

  &__container {
    flex-direction: column;
    align-items: center;

    @media ($medium-up) {
      width: 100%;
      flex-flow: row wrap;
      align-items: flex-start;
      justify-content: space-between;
    }
  }

  &__Locations-container {
    align-items: center;

    @media ($medium-up) {
      align-items: start;
      width: 50%;
    }
  }

  &__Regions-container {
    @media ($medium-up) {
      max-height: spacing(180);
    }
  }

  &__Regions-content {
    align-items: center;

    @media ($medium-up) {
      align-items: flex-start;
      margin-right: spacing(16);
    }
  }

  &__Regions-store {
    text-align: center;

    @media ($medium-up) {
      text-align: left;
    }
  }

  &__Links-container {
    align-items: center;
    margin: 0;

    @media ($medium-up) {
      align-items: flex-start;
      margin: spacing(16);
    }
  }

  &__arrow {
    margin-top: spacing(-10);
  }

  &__privacy-link:hover {
    color: color('goldenrod');
  }

  &__illustration {
    position: absolute;
    bottom: spacing(16);
    right: spacing(32);
    width: 14%;
  }
}
