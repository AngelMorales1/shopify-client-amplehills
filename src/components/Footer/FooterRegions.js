import React, { Component } from 'react';
import { sortHours, abbreviateDay } from 'utils/sortHours';
import styles from './Footer.scss';

class FooterRegions extends Component {
  render() {
    return (
      <div className="mx3">
        <h3 className={`footer-text my2 ${styles['sub-title-text']}`}>
          {this.props.region}
        </h3>
        {this.props.stores.map(store => {
          let hours = sortHours(store.fields);
          return (
            <div className="mb3" key={store.sys.id}>
              <h4 className={`footer-text mb1 ${styles['content-title-text']}`}>
                {store.fields.title}
              </h4>
              {Object.keys(hours).map((hour, idx) => {
                return (
                  <p
                    className={`footer-text ${styles['content-text']} mb1`}
                    key={idx}
                  >{`${hours[hour]}: ${hour}`}</p>
                );
              })}
              {store.fields.delivery ? (
                <button className={`${styles['delivery-button']}`}>
                  ORDER DELIVERY
                </button>
              ) : null}
            </div>
          );
        })}
      </div>
    );
  }
}

export default FooterRegions;
