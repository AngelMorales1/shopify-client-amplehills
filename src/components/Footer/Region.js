import React, { Component } from 'react';

class Region extends Component {
  sortHours(fields) {
    let weeksInOrder = [
      'sunday',
      'monday',
      'tuesday',
      'wednesday',
      'thursday',
      'friday',
      'saturday'
    ];

    //sort data to have only open hours
    let openHours = Object.keys(fields)
      .filter(field => field.includes('day'))
      .reduce((acc, cur) => {
        acc[cur] = fields[cur];
        return acc;
      }, {});

    //sorting by same open hours
    let sortByOpenHours = weeksInOrder.reduce((acc, cur) => {
      let time = openHours[cur];
      acc[time] ? (acc[time] = acc[time].concat([cur])) : (acc[time] = [cur]);
      return acc;
    }, {});

    let sortAsPeriod = Object.keys(sortByOpenHours).reduce((acc, cur) => {
      let days = sortByOpenHours[cur];

      if (
        days.includes(weeksInOrder[0]) &&
        days.includes(weeksInOrder[weeksInOrder.length - 1]) &&
        days.length < 7
      ) {
        for (let i = 0; i < days.length; i++) {
          if (
            weeksInOrder.indexOf(days[i + 1]) -
              weeksInOrder.indexOf(days[i]) !==
            1
          ) {
            acc[cur] = `${this.makeFirstLetterUppercase(
              days[i + 1]
            )} - ${this.makeFirstLetterUppercase(days[0])}`;
            return acc;
          }
        }
      } else if (days.length === 7) {
        acc[cur] = 'Everyday';
        return acc;
      }
      days.length > 1
        ? (acc[cur] = `${this.makeFirstLetterUppercase(
            days[0]
          )} - ${this.makeFirstLetterUppercase(days[days.length - 1])}`)
        : (acc[cur] = this.makeFirstLetterUppercase(days[0]));
      return acc;
    }, {});

    return sortAsPeriod;
  }

  makeFirstLetterUppercase(str) {
    return str[0].toUpperCase() + str.slice(1, 3);
  }

  render() {
    return (
      <div className="border">
        <h1>
          <strong>{this.props.region}</strong>
        </h1>
        {this.props.stores.map(store => {
          let hours = this.sortHours(store.fields);
          return (
            <div key={store.sys.id}>
              {store.fields.title}
              {Object.keys(hours).map((hour, idx) => {
                return <div key={idx}>{`${hours[hour]}: ${hour}`}</div>;
              })}
              {store.fields.delivery ? <button>ORDER DELIVERY</button> : null}
            </div>
          );
        })}
      </div>
    );
  }
}

export default Region;
