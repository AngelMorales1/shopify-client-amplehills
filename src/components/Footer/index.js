import React, { Component } from "react";

class Footer extends Component {
  render() {
    return <span>&copy; 2018 Ample Hills</span>;
  }
}

export default Footer;
