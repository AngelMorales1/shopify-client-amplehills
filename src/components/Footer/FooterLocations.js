import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

import { sortHours } from 'utils/sortHours';
import FooterRegions from 'constants/FooterRegions';
import locationModel from 'models/locationModel';

import styles from './Footer.scss';

const FooterLocations = ({ locations }) => {
  const locationsSortedByRegion = (locations = []) => {
    const regions = locations.reduce((accumulated, current) => {
      let region = current.region;
      accumulated[region] = accumulated[region]
        ? accumulated[region].concat([current])
        : [current];

      return accumulated;
    }, {});

    return FooterRegions.reduce(
      (accumulated, region) => ({
        ...accumulated,
        [region]: regions[region]
      }),
      {}
    );
  };

  const regions = locationsSortedByRegion(locations);

  return (
    <div
      className={cx(
        'flex flex-column m4',
        styles['Footer__Locations-container']
      )}
    >
      <h2 className="mb2 text-white block-headline">Locations</h2>
      <div
        className={cx(
          'flex flex-column flex-wrap',
          styles['Footer__Regions-container']
        )}
      >
        {Object.keys(regions).map(region => (
          <div
            key={region}
            className={cx(
              'flex flex-column',
              styles['Footer__Regions-content']
            )}
          >
            <h3 className="my2 text-white callout">{region}</h3>
            {regions[region].map(location => {
              let hours = sortHours(location.hours);

              return (
                <div
                  className={cx('mb3', styles['Footer__Regions-store'])}
                  key={location.id}
                >
                  <h4 className="mb1 text-white bold small nowrap">
                    {location.title}
                  </h4>
                  {hours.map((hour, i) => {
                    const hourKey = Object.keys(hour);
                    return (
                      <p
                        className="mb1 text-white small nowrap"
                        key={i}
                      >{`${hourKey}: ${hour[hourKey]}`}</p>
                    );
                  })}
                  {location.delivery ? (
                    <div className="bg-white text-madison-blue inline-block mt1 nowrap tag">
                      Order Delivery
                    </div>
                  ) : null}
                </div>
              );
            })}
          </div>
        ))}
      </div>
    </div>
  );
};

export default FooterLocations;

FooterLocations.propTypes = {
  locations: PropTypes.arrayOf(locationModel.propTypes)
};

FooterLocations.defaultProps = {
  locations: []
};
