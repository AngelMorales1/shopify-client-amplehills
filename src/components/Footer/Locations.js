import React, { Component } from 'react';
import Region from './Region.js';

class Locations extends Component {
  render() {
    let regions = Object.keys(this.props.locations);

    return (
      <div>
        {regions.map(region => (
          <Region
            key={region}
            region={region}
            stores={this.props.locations[region]}
          />
        ))}
      </div>
    );
  }
}

export default Locations;
