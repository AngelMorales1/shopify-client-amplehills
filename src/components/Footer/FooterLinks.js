import React, { Component } from 'react';
import cx from 'classnames';
import styles from './Footer.scss';
import { Button, Image } from 'components/base';

class FooterLinks extends Component {
  render() {
    return (
      <div className="m4 col-4">
        <h2 className="title bold text-white">We would love to talk!</h2>
        <div className="my3 flex">
          <Button label="Contact Us" color="white-madison-blue" />
          <Image
            alt="Decorative arrow icon"
            src="/assets/images/arrow-left-down.svg"
            className={cx('arrow mx3', styles['Footer__arrow'])}
          />
        </div>
        <div className="mt4">
          <div className="my3 flex items-center">
            <Image
              alt="Instagram icon"
              src="/assets/images/bubble-icon.svg"
              className="icon"
            />
            <a
              href={this.props.footerLinks.instagramLink}
              target="_blank"
              rel="noopener"
              className="bold text-white text-decoration-none ml3"
            >
              Instagram
            </a>
          </div>
          <div className="my3 flex items-center">
            <Image
              alt="Instagram icon"
              src="/assets/images/bubble-icon.svg"
              className="icon"
            />
            <a
              href={this.props.footerLinks.twitterLink}
              target="_blank"
              rel="noopener"
              className="bold text-white text-decoration-none ml3"
            >
              Twitter
            </a>
          </div>
          <div className="my3 flex items-center">
            <Image
              alt="Instagram icon"
              src="/assets/images/bubble-icon.svg"
              className="icon"
            />
            <a
              href={this.props.footerLinks.facebookLink}
              target="_blank"
              rel="noopener"
              className="bold text-white text-decoration-none ml3"
            >
              Facebook
            </a>
          </div>
        </div>
      </div>
    );
  }
}

export default FooterLinks;
