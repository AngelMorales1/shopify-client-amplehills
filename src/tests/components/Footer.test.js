import React from 'react';
import { shallow } from 'enzyme';

import Footer from 'components/Footer';

it('renders without data', () => {
  const component = shallow(<Footer />);

  expect(component).toMatchSnapshot();
});

it('renders with data', () => {
  const component = shallow(
    <Footer
      footerIllustration={{
        fields: {
          description: 'foo',
          file: {
            contentType: 'foo',
            details: { size: 0, image: {} },
            fileName: 'foo',
            url: '/foo'
          },
          title: 'foo'
        }
      }}
      footerLinks={{
        facebookLink: 'foo',
        instagramLink: 'foo',
        twitterLink: 'foo'
      }}
      locations={[
        {
          id: 'foo',
          title: 'foo',
          image: 'foo',
          seasonalImage: 'foo',
          address1: 'foo',
          address2: 'foo',
          city: 'foo',
          region: 'foo',
          state: 'foo',
          zip: 'foo',
          coordinates: {
            lon: 0,
            lat: 0
          },
          phone: 'foo',
          seasonal: false,
          hours: {
            monday: 'foo',
            tuesday: 'foo',
            wednesday: 'foo',
            thursday: 'foo',
            friday: 'foo',
            saturday: 'foo',
            sunday: 'foo'
          },
          sortedHours: [{ foo: 'foo' }],
          delivery: false
        }
      ]}
    />
  );

  expect(component).toMatchSnapshot();
});
