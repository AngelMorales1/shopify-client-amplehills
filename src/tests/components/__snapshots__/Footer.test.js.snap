// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with data 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Footer
    footerIllustration={
        Object {
            "fields": Object {
              "description": "foo",
              "file": Object {
                "contentType": "foo",
                "details": Object {
                  "image": Object {},
                  "size": 0,
                },
                "fileName": "foo",
                "url": "/foo",
              },
              "title": "foo",
            },
          }
    }
    footerLinks={
        Object {
            "facebookLink": "foo",
            "instagramLink": "foo",
            "twitterLink": "foo",
          }
    }
    locations={
        Array [
            Object {
              "address1": "foo",
              "address2": "foo",
              "city": "foo",
              "coordinates": Object {
                "lat": 0,
                "lon": 0,
              },
              "delivery": false,
              "hours": Object {
                "friday": "foo",
                "monday": "foo",
                "saturday": "foo",
                "sunday": "foo",
                "thursday": "foo",
                "tuesday": "foo",
                "wednesday": "foo",
              },
              "id": "foo",
              "image": "foo",
              "phone": "foo",
              "region": "foo",
              "seasonal": false,
              "seasonalImage": "foo",
              "sortedHours": Array [
                Object {
                  "foo": "foo",
                },
              ],
              "state": "foo",
              "title": "foo",
              "zip": "foo",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="flex"
>
          <FooterLocations
                    locations={
                              Array [
                                        Object {
                                          "address1": "foo",
                                          "address2": "foo",
                                          "city": "foo",
                                          "coordinates": Object {
                                            "lat": 0,
                                            "lon": 0,
                                          },
                                          "delivery": false,
                                          "hours": Object {
                                            "friday": "foo",
                                            "monday": "foo",
                                            "saturday": "foo",
                                            "sunday": "foo",
                                            "thursday": "foo",
                                            "tuesday": "foo",
                                            "wednesday": "foo",
                                          },
                                          "id": "foo",
                                          "image": "foo",
                                          "phone": "foo",
                                          "region": "foo",
                                          "seasonal": false,
                                          "seasonalImage": "foo",
                                          "sortedHours": Array [
                                            Object {
                                              "foo": "foo",
                                            },
                                          ],
                                          "state": "foo",
                                          "title": "foo",
                                          "zip": "foo",
                                        },
                                      ]
                    }
          />
          <FooterLinks
                    facebookLink=""
                    footerIllustration={
                              Object {
                                        "fields": Object {
                                          "description": "foo",
                                          "file": Object {
                                            "contentType": "foo",
                                            "details": Object {
                                              "image": Object {},
                                              "size": 0,
                                            },
                                            "fileName": "foo",
                                            "url": "/foo",
                                          },
                                          "title": "foo",
                                        },
                                      }
                    }
                    footerLinks={
                              Object {
                                        "facebookLink": "foo",
                                        "instagramLink": "foo",
                                        "twitterLink": "foo",
                                      }
                    }
                    instagramLink=""
                    twitterLink=""
          />
</div>,
        <span
          className="bold small text-white center"
>
          Â© 2018 Ample Hills Creamery. Privacy Policy & Accessibility
</span>,
        <Image
          alt=""
          bg={false}
          className="medium-down-display-none xs-hide sm-hide"
          onImgLoad={[Function]}
          src="/foo"
          style={Object {}}
/>,
      ],
      "className": "flex flex-column p4 bg-madison-blue",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <FooterLocations
              locations={
                            Array [
                                          Object {
                                            "address1": "foo",
                                            "address2": "foo",
                                            "city": "foo",
                                            "coordinates": Object {
                                              "lat": 0,
                                              "lon": 0,
                                            },
                                            "delivery": false,
                                            "hours": Object {
                                              "friday": "foo",
                                              "monday": "foo",
                                              "saturday": "foo",
                                              "sunday": "foo",
                                              "thursday": "foo",
                                              "tuesday": "foo",
                                              "wednesday": "foo",
                                            },
                                            "id": "foo",
                                            "image": "foo",
                                            "phone": "foo",
                                            "region": "foo",
                                            "seasonal": false,
                                            "seasonalImage": "foo",
                                            "sortedHours": Array [
                                              Object {
                                                "foo": "foo",
                                              },
                                            ],
                                            "state": "foo",
                                            "title": "foo",
                                            "zip": "foo",
                                          },
                                        ]
              }
/>,
            <FooterLinks
              facebookLink=""
              footerIllustration={
                            Object {
                                          "fields": Object {
                                            "description": "foo",
                                            "file": Object {
                                              "contentType": "foo",
                                              "details": Object {
                                                "image": Object {},
                                                "size": 0,
                                              },
                                              "fileName": "foo",
                                              "url": "/foo",
                                            },
                                            "title": "foo",
                                          },
                                        }
              }
              footerLinks={
                            Object {
                                          "facebookLink": "foo",
                                          "instagramLink": "foo",
                                          "twitterLink": "foo",
                                        }
              }
              instagramLink=""
              twitterLink=""
/>,
          ],
          "className": "flex",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "locations": Array [
                Object {
                  "address1": "foo",
                  "address2": "foo",
                  "city": "foo",
                  "coordinates": Object {
                    "lat": 0,
                    "lon": 0,
                  },
                  "delivery": false,
                  "hours": Object {
                    "friday": "foo",
                    "monday": "foo",
                    "saturday": "foo",
                    "sunday": "foo",
                    "thursday": "foo",
                    "tuesday": "foo",
                    "wednesday": "foo",
                  },
                  "id": "foo",
                  "image": "foo",
                  "phone": "foo",
                  "region": "foo",
                  "seasonal": false,
                  "seasonalImage": "foo",
                  "sortedHours": Array [
                    Object {
                      "foo": "foo",
                    },
                  ],
                  "state": "foo",
                  "title": "foo",
                  "zip": "foo",
                },
              ],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "facebookLink": "",
              "footerIllustration": Object {
                "fields": Object {
                  "description": "foo",
                  "file": Object {
                    "contentType": "foo",
                    "details": Object {
                      "image": Object {},
                      "size": 0,
                    },
                    "fileName": "foo",
                    "url": "/foo",
                  },
                  "title": "foo",
                },
              },
              "footerLinks": Object {
                "facebookLink": "foo",
                "instagramLink": "foo",
                "twitterLink": "foo",
              },
              "instagramLink": "",
              "twitterLink": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Â© 2018 Ample Hills Creamery. Privacy Policy & Accessibility",
          "className": "bold small text-white center",
        },
        "ref": null,
        "rendered": "Â© 2018 Ample Hills Creamery. Privacy Policy & Accessibility",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "alt": "",
          "bg": false,
          "children": null,
          "className": "medium-down-display-none xs-hide sm-hide",
          "onImgLoad": [Function],
          "src": "/foo",
          "style": Object {},
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="flex"
>
            <FooterLocations
                        locations={
                                    Array [
                                                Object {
                                                  "address1": "foo",
                                                  "address2": "foo",
                                                  "city": "foo",
                                                  "coordinates": Object {
                                                    "lat": 0,
                                                    "lon": 0,
                                                  },
                                                  "delivery": false,
                                                  "hours": Object {
                                                    "friday": "foo",
                                                    "monday": "foo",
                                                    "saturday": "foo",
                                                    "sunday": "foo",
                                                    "thursday": "foo",
                                                    "tuesday": "foo",
                                                    "wednesday": "foo",
                                                  },
                                                  "id": "foo",
                                                  "image": "foo",
                                                  "phone": "foo",
                                                  "region": "foo",
                                                  "seasonal": false,
                                                  "seasonalImage": "foo",
                                                  "sortedHours": Array [
                                                    Object {
                                                      "foo": "foo",
                                                    },
                                                  ],
                                                  "state": "foo",
                                                  "title": "foo",
                                                  "zip": "foo",
                                                },
                                              ]
                        }
            />
            <FooterLinks
                        facebookLink=""
                        footerIllustration={
                                    Object {
                                                "fields": Object {
                                                  "description": "foo",
                                                  "file": Object {
                                                    "contentType": "foo",
                                                    "details": Object {
                                                      "image": Object {},
                                                      "size": 0,
                                                    },
                                                    "fileName": "foo",
                                                    "url": "/foo",
                                                  },
                                                  "title": "foo",
                                                },
                                              }
                        }
                        footerLinks={
                                    Object {
                                                "facebookLink": "foo",
                                                "instagramLink": "foo",
                                                "twitterLink": "foo",
                                              }
                        }
                        instagramLink=""
                        twitterLink=""
            />
</div>,
          <span
            className="bold small text-white center"
>
            Â© 2018 Ample Hills Creamery. Privacy Policy & Accessibility
</span>,
          <Image
            alt=""
            bg={false}
            className="medium-down-display-none xs-hide sm-hide"
            onImgLoad={[Function]}
            src="/foo"
            style={Object {}}
/>,
        ],
        "className": "flex flex-column p4 bg-madison-blue",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <FooterLocations
                locations={
                                Array [
                                                Object {
                                                  "address1": "foo",
                                                  "address2": "foo",
                                                  "city": "foo",
                                                  "coordinates": Object {
                                                    "lat": 0,
                                                    "lon": 0,
                                                  },
                                                  "delivery": false,
                                                  "hours": Object {
                                                    "friday": "foo",
                                                    "monday": "foo",
                                                    "saturday": "foo",
                                                    "sunday": "foo",
                                                    "thursday": "foo",
                                                    "tuesday": "foo",
                                                    "wednesday": "foo",
                                                  },
                                                  "id": "foo",
                                                  "image": "foo",
                                                  "phone": "foo",
                                                  "region": "foo",
                                                  "seasonal": false,
                                                  "seasonalImage": "foo",
                                                  "sortedHours": Array [
                                                    Object {
                                                      "foo": "foo",
                                                    },
                                                  ],
                                                  "state": "foo",
                                                  "title": "foo",
                                                  "zip": "foo",
                                                },
                                              ]
                }
/>,
              <FooterLinks
                facebookLink=""
                footerIllustration={
                                Object {
                                                "fields": Object {
                                                  "description": "foo",
                                                  "file": Object {
                                                    "contentType": "foo",
                                                    "details": Object {
                                                      "image": Object {},
                                                      "size": 0,
                                                    },
                                                    "fileName": "foo",
                                                    "url": "/foo",
                                                  },
                                                  "title": "foo",
                                                },
                                              }
                }
                footerLinks={
                                Object {
                                                "facebookLink": "foo",
                                                "instagramLink": "foo",
                                                "twitterLink": "foo",
                                              }
                }
                instagramLink=""
                twitterLink=""
/>,
            ],
            "className": "flex",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "locations": Array [
                  Object {
                    "address1": "foo",
                    "address2": "foo",
                    "city": "foo",
                    "coordinates": Object {
                      "lat": 0,
                      "lon": 0,
                    },
                    "delivery": false,
                    "hours": Object {
                      "friday": "foo",
                      "monday": "foo",
                      "saturday": "foo",
                      "sunday": "foo",
                      "thursday": "foo",
                      "tuesday": "foo",
                      "wednesday": "foo",
                    },
                    "id": "foo",
                    "image": "foo",
                    "phone": "foo",
                    "region": "foo",
                    "seasonal": false,
                    "seasonalImage": "foo",
                    "sortedHours": Array [
                      Object {
                        "foo": "foo",
                      },
                    ],
                    "state": "foo",
                    "title": "foo",
                    "zip": "foo",
                  },
                ],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "facebookLink": "",
                "footerIllustration": Object {
                  "fields": Object {
                    "description": "foo",
                    "file": Object {
                      "contentType": "foo",
                      "details": Object {
                        "image": Object {},
                        "size": 0,
                      },
                      "fileName": "foo",
                      "url": "/foo",
                    },
                    "title": "foo",
                  },
                },
                "footerLinks": Object {
                  "facebookLink": "foo",
                  "instagramLink": "foo",
                  "twitterLink": "foo",
                },
                "instagramLink": "",
                "twitterLink": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Â© 2018 Ample Hills Creamery. Privacy Policy & Accessibility",
            "className": "bold small text-white center",
          },
          "ref": null,
          "rendered": "Â© 2018 Ample Hills Creamery. Privacy Policy & Accessibility",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "alt": "",
            "bg": false,
            "children": null,
            "className": "medium-down-display-none xs-hide sm-hide",
            "onImgLoad": [Function],
            "src": "/foo",
            "style": Object {},
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`renders without data 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Footer
    footerIllustration={
        Object {
            "fields": Object {
              "description": "",
              "file": Object {
                "title": "",
                "url": "",
              },
            },
          }
    }
    footerLinks={Object {}}
    locations={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="flex"
>
          <FooterLocations
                    locations={Array []}
          />
          <FooterLinks
                    facebookLink=""
                    footerIllustration={
                              Object {
                                        "fields": Object {
                                          "description": "",
                                          "file": Object {
                                            "title": "",
                                            "url": "",
                                          },
                                        },
                                      }
                    }
                    footerLinks={Object {}}
                    instagramLink=""
                    twitterLink=""
          />
</div>,
        <span
          className="bold small text-white center"
>
          Â© 2018 Ample Hills Creamery. Privacy Policy & Accessibility
</span>,
        <Image
          alt=""
          bg={false}
          className="medium-down-display-none xs-hide sm-hide"
          onImgLoad={[Function]}
          src=""
          style={Object {}}
/>,
      ],
      "className": "flex flex-column p4 bg-madison-blue",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <FooterLocations
              locations={Array []}
/>,
            <FooterLinks
              facebookLink=""
              footerIllustration={
                            Object {
                                          "fields": Object {
                                            "description": "",
                                            "file": Object {
                                              "title": "",
                                              "url": "",
                                            },
                                          },
                                        }
              }
              footerLinks={Object {}}
              instagramLink=""
              twitterLink=""
/>,
          ],
          "className": "flex",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "locations": Array [],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "facebookLink": "",
              "footerIllustration": Object {
                "fields": Object {
                  "description": "",
                  "file": Object {
                    "title": "",
                    "url": "",
                  },
                },
              },
              "footerLinks": Object {},
              "instagramLink": "",
              "twitterLink": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Â© 2018 Ample Hills Creamery. Privacy Policy & Accessibility",
          "className": "bold small text-white center",
        },
        "ref": null,
        "rendered": "Â© 2018 Ample Hills Creamery. Privacy Policy & Accessibility",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "alt": "",
          "bg": false,
          "children": null,
          "className": "medium-down-display-none xs-hide sm-hide",
          "onImgLoad": [Function],
          "src": "",
          "style": Object {},
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="flex"
>
            <FooterLocations
                        locations={Array []}
            />
            <FooterLinks
                        facebookLink=""
                        footerIllustration={
                                    Object {
                                                "fields": Object {
                                                  "description": "",
                                                  "file": Object {
                                                    "title": "",
                                                    "url": "",
                                                  },
                                                },
                                              }
                        }
                        footerLinks={Object {}}
                        instagramLink=""
                        twitterLink=""
            />
</div>,
          <span
            className="bold small text-white center"
>
            Â© 2018 Ample Hills Creamery. Privacy Policy & Accessibility
</span>,
          <Image
            alt=""
            bg={false}
            className="medium-down-display-none xs-hide sm-hide"
            onImgLoad={[Function]}
            src=""
            style={Object {}}
/>,
        ],
        "className": "flex flex-column p4 bg-madison-blue",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <FooterLocations
                locations={Array []}
/>,
              <FooterLinks
                facebookLink=""
                footerIllustration={
                                Object {
                                                "fields": Object {
                                                  "description": "",
                                                  "file": Object {
                                                    "title": "",
                                                    "url": "",
                                                  },
                                                },
                                              }
                }
                footerLinks={Object {}}
                instagramLink=""
                twitterLink=""
/>,
            ],
            "className": "flex",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "locations": Array [],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "facebookLink": "",
                "footerIllustration": Object {
                  "fields": Object {
                    "description": "",
                    "file": Object {
                      "title": "",
                      "url": "",
                    },
                  },
                },
                "footerLinks": Object {},
                "instagramLink": "",
                "twitterLink": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Â© 2018 Ample Hills Creamery. Privacy Policy & Accessibility",
            "className": "bold small text-white center",
          },
          "ref": null,
          "rendered": "Â© 2018 Ample Hills Creamery. Privacy Policy & Accessibility",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "alt": "",
            "bg": false,
            "children": null,
            "className": "medium-down-display-none xs-hide sm-hide",
            "onImgLoad": [Function],
            "src": "",
            "style": Object {},
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
